app/src/androidTest/java/com/sdp/movemeet/view/profile/EditProfileActivityImagePickerTest.java

```diff
@@ -21,46 +30,83 @@
 import java.io.FileNotFoundException;
 import java.io.FileOutputStream;
 import java.io.IOException;
+import java.util.concurrent.CountDownLatch;
 
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
 import static junit.framework.Assert.assertTrue;
 
+// This test tests both ImageHandler.uploadImage and ImageHandler.loadImage since the later one is
+// an integral part of the first one.
 public class EditProfileActivityImagePickerTest {
 
-    private StorageReference storageReference;
-    private String imagePath = "users/ydXLPxZQMXRRVHY4oGUTggDpZvp1/profile.jpg";
+    private String userImagePath = "users/B6PwbuQT3rRIyOemWtNQu7xfXmq2/profile.jpg"; // Antho1
+
+    private FirebaseAuth fAuth;
 
     @Rule
     public IntentsTestRule<EditProfileActivity> activityRule = new IntentsTestRule<>(EditProfileActivity.class);
 
     @Before
     public void setUp() {
-        // Setting up Firebase Storage
-        storageReference = FirebaseStorage.getInstance().getReference();
-
         // Saving the mocked picked image
         savePickedImage();
+
+        CountDownLatch latch = new CountDownLatch(1);
+
+        fAuth = AuthenticationInstanceProvider.getAuthenticationInstance();
+        fAuth.signInWithEmailAndPassword("antho1@gmail.com", "123456").addOnSuccessListener(new OnSuccessListener<AuthResult>() {
+            @Override
+            public void onSuccess(AuthResult authResult) {
+                latch.countDown();
+            }
+        }).addOnFailureListener(new OnFailureListener() {
+            @Override
+            public void onFailure(@NonNull Exception e) {
+                assert(false);
+            }
+        });
+
+        try {
+            latch.await();
+        } catch (InterruptedException e) {
+            assert(false);
+        }
+
+        ActivityScenario scenario = ActivityScenario.launch(EditProfileActivity.class);
+
     }
 
     @Test
     public void uploadImageToFirebaseStorage() {
         File dir = activityRule.getActivity().getExternalCacheDir();
-        File file = new File(dir.getPath(), "pickImageResult.jpeg");
+        File file = new File(dir.getPath(), "pickImageResult.png");
```

Comment:
The conversion for other formats than .png is not possible ?

Reply:
I guess it is, but I noticed that it works well with `.png` images and I stick to it! In fact, it matches the method I used below `bm.compress(Bitmap.CompressFormat.PNG, 100, outStream);` and the image I choose to set is a .png picture: icon_tricking.png.